src/external/libraries/open_cv_types.hpp

```diff
@@ -10,21 +10,27 @@
 #ifndef OPEN_CV_TYPES_HPP
 #define OPEN_CV_TYPES_HPP
 
-#include <vector>
-#include "open_cv.hpp"
 #include "../../image/sample_image.hpp"
+#include "open_cv.hpp"
+#include <vector>
 
-namespace lucidy{
-    typedef std::vector<cv::KeyPoint> FeatureList;
-    typedef cv::Ptr<cv::xfeatures2d::SURF> SurfDetector;
-    typedef cv::Mat DescriptorList;
-    typedef struct{int x; int y;} Vector;
-    typedef std::vector<cv::DMatch> MatchList;
-    typedef cv::Mat AffineMatrix;
-
-    struct FoundImageData{ SampleImage image; AffineMatrix matrix; bool match; };
-    typedef std::vector<FoundImageData> FoundImageList;
+namespace lucidy {
+typedef std::vector<cv::KeyPoint> FeatureList;
+typedef cv::Ptr<cv::xfeatures2d::SURF> SurfDetector;
+typedef cv::Mat DescriptorList;
+typedef struct {
```

Comment:
Not documented with Doxygen

typedef struct is C style struct declarations. Use classes